rsplan/primitives.py

```diff
@@ -309,8 +308,9 @@ def _interpolated(
         # straight segments) and dtheta (step size / turn radius) for curved segments.
         step = step_size if self.is_straight else step_size / self.turn_radius
 
-
+        # Calculate num_steps to guarantee at least 2 points (start and end),
+        # addressing cases where magnitude < step size.
+        num_steps = int((magnitude / step) + 2)
```

Comment:
Any way we could DRY this out with the similar calculation on `_straight_runway_pts`?